### DISCUSSION TITLE
OP_RETURN_DATA missing in outputs section

### DISCUSSION THREAD

**Discussion by gomesalexandre** [phantom/docs#295] [Help & Support] [answered]
### Summary

gm!

Managed to sign and broadcast a BTC Tx with OP_RETURN_DATA: https://live.blockcypher.com/btc/tx/d0f6d74c8ad3c5fda41d1e6fe0b5d2256524b247b4558a643786df910cf7bf2e/

While OP_RETURN_DATA is indeed present as an output, Phantom fails to parse it and only parses the two other outputs in this Tx (i.e change/spend outputs in this screenshot respectively):

<img width="1457" alt="Screenshot 2024-09-20 at 16 40 39" src="https://github.com/user-attachments/assets/936152bc-afc5-46b1-8eff-25d9f7e3b666">

The OP_RETURN_DATA output itself is build in a pretty standard way with bitcoinjs-lib:

```typescript
    const data = Buffer.from(msg.opReturnData, "utf-8");
    const embed = bitcoin.payments.embed({ data: [data] });
    psbt.addOutput({
      script: embed.output!,
      value: 0,
    });
```

Not sure if this is so much of a bug rather than an unimplemented feature, but I believe it would be nice to see OP_RETURN_DATA outputs in the expanded section 🙏🏽 

### Steps to Reproduce

- Build a UTXO Tx containing an OP_RETURN_DATA output and sign with `signPSBT()` method on the bitcoin Phantom provider
- Notice data output isn't present in outputs

### Is there an existing discussion for this?

- [X] I have searched the existing discussions

**Comment by adamdelphantom** [comment]
@gomesalexandre what would be your use case for the output?

**Reply by gomesalexandre** [reply, marked as answer]
Either sends with a memo of sorts or dApps (e.g THORChain) leveraging BTC memos, would be a nice feature to see it!
On top of seeing the outputs (minus the `OP_RETURN_DATA` one), which is already quite an unusual and killer feature for BTC UX 💯
